tensorflow/python/ops/array_ops.py

```diff
@@ -1085,6 +1085,7 @@ def boolean_mask(tensor, mask, name="boolean_mask"):
     tensor:  N-D tensor.
     mask:  K-D boolean tensor, K <= N and K must be known statically.
     name:  A name for this operation (optional).
+    axis: The axis in `tensor` to mask from. Defaults to the first dimension.
```

Comment:
`boolean_mask(tensor, mask)` is supposed to act like numpy's `my_array[my_mask]`, which it currently does.  What you're proposing here looks like [`numpy.compress`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.compress.html#numpy.compress)

So it seems like this functionality should be part of an `array_ops.compress`.